src/ng/q.js

```diff
@@ -361,21 +350,27 @@ function qFactory(nextTick, exceptionHandler) {
     },
 
     $$resolve: function(val) {
-      var then, fns;
+      if (this.promise.$$state.status > 0) return;
 
-      fns = callOnce(this, this.$$resolve, this.$$reject);
+      if (isPromiseLike(val)) {
```

Comment:
also, `true.then` (if defined), must not be handled as a promise (same with numeric values)